datafusion/sql/src/expr/arrow_cast.rs

```diff
@@ -673,10 +748,21 @@ mod test {
             ("", "Error finding next token"),
             ("null", "Unsupported type 'null'"),
             ("Nu", "Unsupported type 'Nu'"),
-            // TODO support timezones
             (
-                r#"Timestamp(Nanosecond, Some("UTC"))"#,
-                "Error unrecognized word: Some",
+                r#"Timestamp(Nanosecond, Some(+00:00))"#,
+                "Error unrecognized word: +00:00",
+            ),
+            (
+                r#"Timestamp(Nanosecond, Some("+00:00))"#,
+                r#"parsing "+00:00 as double quoted string: last char must be ""#,
+            ),
+            (
+                r#"Timestamp(Nanosecond, Some(""))"#,
+                r#"parsing "" as double quoted string: empty string isn't supported"#,
+            ),
+            (
+                r#"Timestamp(Nanosecond, Some("+00:00""))"#,
+                r#"parsing "+00:00"" as double quoted string: escaped double quote isn't supported"#,
```

Comment:
@alamb 
added error cases for 
`empty string` and 
`double quoted string that contains double quote`